samtranslator/model/sam_resources.py

```diff
@@ -2111,23 +2112,32 @@ def _add_connector_metadata(
         original_resources = original_template.get("Resources", {})
         original_source_type = original_resources.get(source.logical_id, {}).get("Type")
         original_dest_type = original_resources.get(destination.logical_id, {}).get("Type")
-        metadata = {
-            "aws:sam:connectors": {
-                self.logical_id: {
-                    # If the source/destination is a serverless resource,
-                    # we prefer to include the original serverless resource type
-                    # over the transformed CFN resource type so it can distinguish
-                    # connector usage between serverless resources and CFN resources.
-                    "Source": {"Type": original_source_type or source.resource_type},
-                    "Destination": {"Type": original_dest_type or destination.resource_type},
-                }
-            }
-        }
+
         for resource in generated_resources:
             # Although as today the generated resources do not have any existing metadata,
             # To make it future proof, we still does a merge to avoid overwriting.
             try:
                 original_metadata = resource.get_resource_attribute("Metadata")
             except KeyError:
                 original_metadata = {}
+
+            connector_type: Dict[str, Any] = {
+                "MultiDestConnector": is_multi_dest,
+                "EmbeddedConnector": "aws:sam:embedded:connector" in original_metadata,
```

Comment:
I still don't get what `aws:sam:embedded:connector` is. It doesn't show up in the final template, but looks misleadingly like something that does.

Reply:
It's used as an intermediate processing step. I intentionally removed it from the final template.

Reply:
Why intermediate?

Reply:
Because the final metadata hasn't been built yet at the time I added `aws:sam:embedded:connector`